site-root: show forges logos on landing page

To1ne opened the pull request:
## Changes

Replace the project and company logos by logos of forges that provide
Git hosting.

SVG images are used to properly scale on any screen size. The SVG images
are embedded directly into the HTML. This allows us to use CSS custom
properties (like `var(--logos-1)`) for the colors. Using this method the
colors can easily adapt to dark/light mode.

The SVG images are created in Inkscape. One source file `all.svg` has
all the logos. In this file placeholder colors are used. The placeholder
colors are replaced by `var(--xyz)` by the Hugo rendering pipeline. The
placeholder colors are:

* `#aaaaaa` -> `var(--logos-1)`
* `#999999` -> `var(--logos-2)`
* `#666666` -> `var(--logos-3)`
* `#333333` -> `var(--logos-4)`

These are lightest to darkest in light mode, and reverse in dark mode.

### TODO

- [x] Add more logos
  - [ ] More from https://en.wikipedia.org/wiki/Forge_(software)#Examples ?
  - [ ] https://tangled.sh/
- [x] Remove color from logos
- [x] Make it work in dark mode
- [x] Shuffle them around randomly

## Context

I don't think it's still relevant to show which projects use Git, because nowadays a lot of projects use Git. Instead link to sites where Git users can host their repositories.

Because we want to be neutral toward any Git forge, I'd like to shuffle the logos around randomly on each page visit.

## Screenshots

![Screen Shot 2025-04-03 at 07 42 06](https://github.com/user-attachments/assets/0ae1ad05-8644-405b-abf8-0b380516ab80)
![Screen Shot 2025-04-03 at 07 41 57](https://github.com/user-attachments/assets/3d2f1bd4-5142-4087-b866-a638e2faf206)

To1ne committed:
images: delete all company project logos

We're about to replace the company project logos on the landing page
with logos of forges. Thus remove the logos we no longer need.

Signed-off-by: Toon Claes <toon@iotcl.com>

To1ne committed:
site-root: show forges logos on landing page

Replace the project and company logos by logos of forges that provide
Git hosting.

SVG images are used to properly scale on any screen size. The SVG images
are embedded directly into the HTML. This allows us to use CSS custom
properties (like `var(--logos-1)`) for the colors. Using this method the
colors can easily adapt to dark/light mode.

The SVG images are created in Inkscape. One source file `all.svg` has
all the logos. In this file placeholder colors are used. The placeholder
colors are replaced by `var(--xyz)` by the Hugo rendering pipeline. The
placeholder colors are:

* `#aaaaaa` -> `var(--logos-1)`
* `#999999` -> `var(--logos-2)`
* `#666666` -> `var(--logos-3)`
* `#333333` -> `var(--logos-4)`

These are lightest to darkest in light mode, and reverse in dark mode.

dscho commented on assets/sass/front-page.scss:
```diff
@@ -200,90 +200,25 @@ $monitor-height: 271px;
       }
-
-      &.google {
-        @include background-image-2x($baseurl + "images/company-project-logos/google", 108px, 65px);
```
Probably makes sense to delete those images, too.

To1ne replied:
I did, didn't I?

dscho replied:
Yes. I missed this somehow, sorry!

dscho reviewed (approved):
I like it! How about merging this PR as-is and letting contributors add more forges in follow-up PRs?

To1ne commented:
@dscho I've fixed the dark mode.

dscho reviewed (approved):
Nice!

To1ne commented:
Even looks decent without CSS:

![image](https://github.com/user-attachments/assets/621b4e3a-0c45-4bc3-987c-d4689b46be11)

To1ne committed:
site-root: randomize the order of the company logos

We don't want to endorse any brand more than any other, so randomize the
logos.

We use Hugo to randomize the order every build. Also in CSS the order is
randomized using a seed and some math. And in JS we set that set every
time the page reloads.

Signed-off-by: Toon Claes <toon@iotcl.com>

To1ne commented:
Currently deploying to https://to1ne.github.io/git-scm.com/. See: https://github.com/To1ne/git-scm.com/actions/runs/14240442557/job/39908921601

To1ne commented:
@dscho demo deploy should be completed. And so is this PR.
